diff --git a/mono/metadata/Makefile.am b/mono/metadata/Makefile.am
index dc9cf2a4d25..5cff6ce4cef 100644
--- a/mono/metadata/Makefile.am
+++ b/mono/metadata/Makefile.am
@@ -387,7 +387,8 @@ common_sources = \
 	callspec.h	\
 	callspec.c	\
 	abi.c	\
-	native-library.c
+	native-library.c \
+	fm-utils.c
 
 # These source files have compile time dependencies on GC code
 gc_dependent_sources = \
@@ -465,7 +466,8 @@ libmonoruntimeinclude_HEADERS = \
 	sgen-bridge.h		\
 	threads.h		\
 	tokentype.h		\
-	verify.h		
+	verify.h	\
+	fm-utils.h	
 
 if !ENABLE_MSVC_ONLY
 
diff --git a/mono/metadata/fm-utils.c b/mono/metadata/fm-utils.c
new file mode 100644
index 00000000000..75919b672cc
--- /dev/null
+++ b/mono/metadata/fm-utils.c
@@ -0,0 +1,14 @@
+#include <config.h>
+#include <mono/metadata/object.h>
+#include <mono/metadata/object-internals.h>
+
+
+MonoObject* mono_fm_delegate_get_target(MonoDelegate *delegate)
+{
+    return delegate->target;
+}
+
+MonoObject* mono_fm_delegate_get_hash(MonoDelegate *delegate)
+{
+    return delegate->delegate_trampoline;
+}
\ No newline at end of file
diff --git a/mono/metadata/fm-utils.h b/mono/metadata/fm-utils.h
new file mode 100644
index 00000000000..d5e3b0472e6
--- /dev/null
+++ b/mono/metadata/fm-utils.h
@@ -0,0 +1,10 @@
+#ifndef __FM_MONO_UTILS_H
+#define __FM_MONO_UTILS_H
+
+#include <mono/metadata/object.h>
+
+
+MonoObject* mono_fm_delegate_get_target(MonoDelegate *delegate);
+MonoObject* mono_fm_delegate_get_hash(MonoDelegate *delegate);
+
+#endif
\ No newline at end of file
diff --git a/mono/mini/CMakeLists.txt b/mono/mini/CMakeLists.txt
index 960b4d8d861..5b052ba6a0a 100644
--- a/mono/mini/CMakeLists.txt
+++ b/mono/mini/CMakeLists.txt
@@ -494,7 +494,9 @@ set(metadata_common_sources
 	method-builder-ilgen.h
 	method-builder-ilgen.c
 	marshal-ilgen.h
-	marshal-ilgen.c)
+	marshal-ilgen.c
+	fm-utils.h
+	fm-utils.c)
 
 set (metadata_gc_dependent_sources
      gc-stats.c
