diff --git a/mono/mini/mini-runtime.c b/mono/mini/mini-runtime.c
index 3b44af7adff..1e34819d81e 100644
--- a/mono/mini/mini-runtime.c
+++ b/mono/mini/mini-runtime.c
@@ -2437,10 +2437,12 @@ lookup_start:
 		}
 
 		if (mono_aot_only) {
-			char *fullname = mono_method_get_full_name (method);
-			mono_error_set_execution_engine (error, "Attempting to JIT compile method '%s' while running in aot-only mode. See https://docs.microsoft.com/xamarin/ios/internals/limitations for more information.\n", fullname);
-			g_free (fullname);
-
+			if (!mono_use_interpreter)
+			{
+				char *fullname = mono_method_get_full_name (method);
+				mono_error_set_execution_engine (error, "Attempting to JIT compile method '%s' while running in aot-only mode. See https://docs.microsoft.com/xamarin/ios/internals/limitations for more information.\n", fullname);
+				g_free (fullname);
+			}
 			return NULL;
 		}
 
