diff --git a/mono/utils/mono-threads-mach-helper.c b/mono/utils/mono-threads-mach-helper.c
index 0a2ff341fc4..c7b5739b097 100644
--- a/mono/utils/mono-threads-mach-helper.c
+++ b/mono/utils/mono-threads-mach-helper.c
@@ -123,7 +123,7 @@ mono_threads_init_dead_letter (void)
 	objectForKey = sel_registerName ("objectForKey:");
 
 	// define the dead letter class
-	mono_dead_letter_class = objc_allocateClassPair (nsobject, "MonoDeadLetter", 0);
+	mono_dead_letter_class = objc_allocateClassPair (nsobject, "FM_MonoDeadLetter", 0);
 	class_addMethod (mono_dead_letter_class, dealloc, (IMP)mono_dead_letter_dealloc, "v@:");
 	objc_registerClassPair (mono_dead_letter_class);
 
@@ -133,7 +133,7 @@ mono_threads_init_dead_letter (void)
 	pool = id_objc_msgSend (id_objc_msgSend (nsautoreleasepool, alloc), init);
 
 	id (*objc_msgSend_char)(id, SEL, const char*) = (id (*)(id, SEL, const char*)) objc_msgSend;
-	mono_dead_letter_key = objc_msgSend_char (nsstring, stringWithUTF8String, "mono-dead-letter");
+	mono_dead_letter_key = objc_msgSend_char (nsstring, stringWithUTF8String, "fm-mono-dead-letter");
 
 	id_objc_msgSend (mono_dead_letter_key, retain);
 	id_objc_msgSend (pool, release);
