diff --git a/mono/utils/mono-dl-windows.c b/mono/utils/mono-dl-windows.c
index 88c4911ee2b..47d26a543ba 100644
--- a/mono/utils/mono-dl-windows.c
+++ b/mono/utils/mono-dl-windows.c
@@ -47,6 +47,7 @@ mono_dl_get_so_suffixes (void)
 void*
 mono_dl_open_file (const char *file, int flags)
 {
+	printf("load dll %s\n", file);
 	gpointer hModule = NULL;
 	if (file) {
 		gunichar2* file_utf16 = g_utf8_to_utf16 (file, strlen (file), NULL, NULL, NULL);
